pkg/state/state.go

```diff
@@ -2822,24 +2823,32 @@ func (st *HelmState) appendValuesControlModeFlag(flags []string, reuseValues boo
 	return flags
 }
 
-func (st *HelmState) appendApiVersionsFlags(flags []string, r *ReleaseSpec, kubeVersion string) []string {
+func (st *HelmState) getApiVersions(r *ReleaseSpec) []string {
 	if len(r.ApiVersions) != 0 {
-		for _, a := range r.ApiVersions {
-			flags = append(flags, "--api-versions", a)
-		}
-	} else {
-		for _, a := range st.ApiVersions {
-			flags = append(flags, "--api-versions", a)
-		}
+		return r.ApiVersions
 	}
+	return st.ApiVersions
+}
 
+func (st *HelmState) getKubeVersion(r *ReleaseSpec, kubeVersion string) string {
 	switch {
 	case kubeVersion != "":
-		flags = append(flags, "--kube-version", kubeVersion)
+		return kubeVersion
 	case r.KubeVersion != "":
-		flags = append(flags, "--kube-version", r.KubeVersion)
+		return r.KubeVersion
 	case st.KubeVersion != "":
-		flags = append(flags, "--kube-version", st.KubeVersion)
+		return st.KubeVersion
+	}
+	return ""
+}
+
+func (st *HelmState) appendApiVersionsFlags(flags []string, r *ReleaseSpec, kubeVersion string) []string {
```

Comment:
add some unittests for this function. please.

Reply:
@xabufr

Reply:
@yxxhero I've added test for `appendApiVersionsFlags`, is it ok for you?

Reply:
LGTM, but why do this PR fix https://github.com/helmfile/helmfile/issues/1623? @xabufr

Reply:
ApiVersion and KubeVersion wasn't correctly injected in Chartify [here](https://github.com/helmfile/helmfile/pull/1624/files#diff-7e4e87a1e777423a3d9a915e00fb78a9c4fc2f4c476863f60b67e89b9c635697R1259), with this PR I keep the same behavior than without chartify: use release or state versions depending what is specified